core/src/elements/ons-ripple/index.js

```diff
@@ -0,0 +1,220 @@
+
+    ['color', 'target', 'center', 'start-radius', 'background'].forEach(e => {
+      this.attributeChangedCallback(e, null, this.getAttribute(e));
+    });
+  }
+
+
+  _compile() {
+    ['_wave', '_background'].forEach(e => {
+      this[e] = document.createElement('div');
+      this[e].classList.add('ripple_' + e);
+      this.appendChild(this[e]);
+    });
+    this.setAttribute('_compiled', '');
+  }
+
+  _calculateCoords(e) {
+    var x, y, h, w, r;
+    var b = this.getBoundingClientRect();
+    if (this._center) {
+      x = b.width / 2;
+      y = b.height / 2;
+      r = Math.sqrt(x * x + y * y);
+    } else {
+      x = (e.clientX || e.changedTouches[0].clientX) - b.left;
+      y = (e.clientY || e.changedTouches[0].clientY) - b.top;
+      h = Math.max(y, b.height - y);
+      w = Math.max(x, b.width - x);
+      r = Math.sqrt(h * h + w * w);
+    }
+    return {x, y, r};
+  }
+
+
+  _rippleAnimation(e, duration = 300) {
+    var
+      {_animator, _wave, _background, _minR} = this,
+      {x, y, r} = this._calculateCoords(e);
+
+    _animator.stopAll({stopNext: 1});
+    _animator.animate(_background, {opacity: 1}, duration);
+
+    util.extend(_wave.style, {
+      opacity: 1,
+      top: y - _minR + 'px',
+      left: x - _minR + 'px',
+      width: 2 * _minR + 'px',
+      height: 2 * _minR + 'px'
+    });
+
+    return _animator.animate(_wave, {
+      // opacity: 1,
+      top: y - r,
+      left: x - r,
+      height: 2 * r,
+      width: 2 * r
+    }, duration);
+  }
+
+
+  _onTap(e){
+    if (!this.isDisabled()) {
+      this._rippleAnimation(e.gesture.srcEvent).then(() => {
+        this._animator.fade(this._wave);
+        this._animator.fade(this._background);
+      });
+    }
+  }
+
+
+  _onHold(e){
+    if (!this.isDisabled()) {
+      this._holding = this._rippleAnimation(e.gesture.srcEvent, 2000);
+
+      this._gestureDetector.on('release', this._onRelease);
+    }
+  }
+
+
+  _onRelease(e){
+    this._holding.speed(300).then(() => {
+      this._animator.stopAll({stopNext: true});
+      this._animator.fade(this._wave);
+      this._animator.fade(this._background);
+    });
+
+    this._holding = false;
+    this._gestureDetector.off('release', this._onRelease);
+  }
+
+
+  _onDragStart(e) {
+    if (this._holding) {
+      return this._onRelease(e);
+    }
+    if (['left', 'right'].indexOf(e.gesture.direction) != -1) {
+      this._onTap(e);
+    }
+  }
+
+
+  attachedCallback() {
+    if (this.getAttribute('target') != 'autofind' &&
+        window.getComputedStyle(this.parentNode).getPropertyValue('position') == 'static') {
+      this.parentNode.style.position = 'relative';
+    }
+
+    if (ons._config.animationsDisabled) {
+      this.setDisabled(true);
+    } else {
+      this._gestureDetector.on('tap', this._onTap);
+      this._gestureDetector.on('hold', this._onHold);
+      this._gestureDetector.on('dragstart', this._onDragStart);
+    }
+  }
+
+
+  detachedCallback() {
```

Comment:
This might be true for all elements, but there seems to be an issue in Chrome where the `detachedCallback` is called before `createdCallback` some times. :(

Creating ~100 elements and detaching I managed to get it happen ~30% of the time...

It looks like conditionally initializing `this._gestureDetector` in `attachedCallback` will solve that problem.

I will try to create a simple example and file a bug report to Chromium. There are lots of issues with the implementation in Chrome unfortunately :(